tests/unit/ViolationUtilsTest.js

```diff
@@ -0,0 +1,505 @@
+import Onyx from 'react-native-onyx';
+import ViolationsUtils from '@libs/Violations/ViolationsUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+describe('getViolationsOnyxData', () => {
+    let transaction;
+    let transactionViolations;
+    let policyRequiresTags;
+    let policyTags;
+    let policyRequiresCategories;
+    let policyCategories;
+
+    beforeEach(() => {
+        transaction = {category: 'Food', tag: 'Lunch', transactionID: '123'};
+        transactionViolations = [
+            {name: 'duplicatedTransaction', type: 'violation', userMessage: ''},
+            {name: 'receiptRequired', type: 'violation', userMessage: ''},
+        ];
+        policyRequiresTags = false;
+        policyTags = {};
+        policyRequiresCategories = false;
+        policyCategories = {Food: {enabled: true}};
+    });
+
+    const execute = () => ViolationsUtils.getViolationsOnyxData(transaction, transactionViolations, policyRequiresTags, policyTags, policyRequiresCategories, policyCategories);
+
+    describe('ordinary operations', () => {
+        it('should return an object with correct shape', () => {
+            // run test
+            const result = execute();
+            // verify results
```

Comment:
Please capitalize comments, and add blank lines before them (unless the comment it in a new level of indentation), per our guidelines. That said, we should remove both of these comments since they don't add any value. In fact, I think you can remove every comment in this file